src/types/dict.rs

```diff
@@ -557,76 +557,73 @@ pub(crate) use borrowed_iter::BorrowedDictIter;
 
 /// Conversion trait that allows a sequence of tuples to be converted into `PyDict`
 /// Primary use case for this trait is `call` and `call_method` methods as keywords argument.
-pub trait IntoPyDict: Sized {
+pub trait IntoPyDict<'py>: Sized {
     /// Converts self into a `PyDict` object pointer. Whether pointer owned or borrowed
     /// depends on implementation.
-    fn into_py_dict(self, py: Python<'_>) -> Bound<'_, PyDict>;
+    fn into_py_dict(self, py: Python<'py>) -> Bound<'_, PyDict>;
```

Comment:
I think we can take the opportunity here to make the return type fallible? AFAIK one of the most likely panics on current conversion traits is unhashable keys.

(This method is new for 0.23 replacing the removed GIL Ref form)

Reply:
That makes sense, I will adapt that 👍